cms/djangoapps/contentstore/rest_api/v0/tests/test_tabs.py

```diff
@@ -0,0 +1,216 @@
+from urllib.parse import urlencode
+
+import ddt
+from django.urls import reverse
+from xmodule.modulestore.tests.factories import ItemFactory
+from xmodule.tabs import CourseTabList
+
+from cms.djangoapps.contentstore.tests.utils import CourseTestCase
+
+
+@ddt.ddt
+class TabsAPITests(CourseTestCase):
+    """Test cases for Tabs (a.k.a Pages) page"""
+
+    def setUp(self):
+        """Common setup for tests"""
+
+        # call super class to setup course, etc.
+        super().setUp()
+
+        # Set the URLs for tests
+        self.url = reverse(
+            "cms.djangoapps.contentstore:v0:course_tab_list",
+            kwargs={"course_id": self.course.id},
+        )
+        self.url_settings = reverse(
+            "cms.djangoapps.contentstore:v0:course_tab_settings",
+            kwargs={"course_id": self.course.id},
+        )
+        self.url_reorder = reverse(
+            "cms.djangoapps.contentstore:v0:course_tab_reorder",
+            kwargs={"course_id": self.course.id},
+        )
+
+        # add a static tab to the course, for code coverage
+        self.test_tab = ItemFactory.create(
+            parent_location=self.course.location,
+            category="static_tab",
+            display_name="Static_1",
+        )
+        self.reload_course()
+
+    def check_invalid_response(self, resp):
+        """Verify response is an error listing the invalid_tab_id"""
+
+        assert resp.status_code, 400
+        print("RESPONSE:", resp.content)
+        resp_content = json.loads(resp.content.decode("utf-8"))
+        assert "developer_message" in resp_content
+        return resp_content["developer_message"]
+
+    def test_reorder_tabs(self):
+        """Test re-ordering of tabs"""
+
+        # get the original tab ids
+        orig_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        tab_ids = list(orig_tab_ids)
+        num_orig_tabs = len(orig_tab_ids)
+
+        # make sure we have enough tabs to play around with
+        assert num_orig_tabs >= 5
+
+        # reorder the last two tabs
+        tab_ids[num_orig_tabs - 1], tab_ids[num_orig_tabs - 2] = tab_ids[num_orig_tabs - 2], tab_ids[num_orig_tabs - 1]
+
+        # remove the middle tab
+        # (the code needs to handle the case where tabs requested for re-ordering is a subset of the tabs in the course)
+        removed_tab = tab_ids.pop(num_orig_tabs // 2)
+        assert len(tab_ids) == num_orig_tabs - 1
+
+        # post the request
+        resp = self.client.post(
+            self.url_reorder,
+            data=[{"tab_id": tab_id} for tab_id in tab_ids],
+            content_type="application/json",
+        )
+        assert resp.status_code == 204
+
+        # reload the course and verify the new tab order
+        self.reload_course()
+        new_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        assert new_tab_ids == tab_ids + [removed_tab]
+        assert new_tab_ids != orig_tab_ids
+
+    def test_reorder_tabs_invalid_list(self):
```

Comment:
Sorry, I am not able to understand this test and why it would fail. Like, I can see there are a couple of things happening here. 

1. Getting course tab ids
2. reordering - swapping 0th and 1st index ids. 
3. posting the changes on the server.
4. asserting error.

Why it would fail? how that's an invalid tab order?

Reply:
It will fail because not all tabs are movable. If you try to move an immovable tab, and error should be raised. 
The first tab is always the course tab, it isn't movable. So an operation trying to move it should fail.

Reply:
Additionally, note that I am reusing the same tests as can be found here: 
https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/tests/test_tabs.py

This is for the original view. So the behaviour should remain the same.

Reply:
Thanks for the explaination. In this context, I would update the docstring in this test to state the exact scenario why this test would fail.